tool/tsh/proxy.go

```diff
@@ -864,6 +867,16 @@ func envVarCommand(format, key, value string) (string, error) {
 	}
 }
 
+// requiresLocalProxyTunnel returns whether the given protocol requires a local proxy with the --tunnel flag.
+func requiresLocalProxyTunnel(protocol string) bool {
+	switch protocol {
+	case defaults.ProtocolSnowflake, defaults.ProtocolDynamoDB, defaults.ProtocolElasticsearch:
```

Comment:
I think you can curl ElastiCache in HTTPS without `--tunnel`. @Tener to confirm.

I assume non-tunnel mode works for DynamoDB as well as long the client uses the right cert and CA? I am ok though if we want to keep it simple by only supporting tunnel mode.

Reply:
Did you mean ElasticSearch?

Reply:
leaving this as-is until @Tener confirms, but I believe that *if* a local proxy is used with elastic search, then it requires the --tunnel flag. See: https://github.com/gravitational/teleport/blob/7aa224e430f708dfb53f365f32871a1e0ea01010/tool/tsh/db.go#L577-L582

Reply:
`tsh db connect` uses `elasticsearch-sql-cli` which supports tunnel mode only. `tsh proxy aws` calls dbcmd's `GetConnectCommandAlternatives()` on the other hand.

Reply:
> I think you can curl Elasticsearch in HTTPS without `--tunnel`. @Tener to confirm.

This is right. For Elasticsearch `--tunnel` is simply much easier for clients. Making curl use custom certs is easy; the same thing is much harder with db clients.

Reply:
fixed